gossipd/gossip.c

```diff
@@ -1522,6 +1522,27 @@ static void setup_listeners(struct daemon *daemon, u16 portnum)
 			      portnum);
 }
 
+/**
+ * Callback function to be called whenever the master connection is idle
+ */
+static bool master_conn_idle(struct io_conn *conn UNUSED,
+			     struct daemon_conn *dc)
+{
+	const u8 *msg, *err;
+	struct daemon *daemon = container_of(dc, struct daemon, master);
+	msg = gossip_store_read_next(tmpctx, daemon->rstate->store);
+
+	if (msg) {
+		err = handle_gossip_msg(daemon, msg);
+
+		/* Since we are replaying from store we should not get errors */
+		assert(!err);
```

Comment:
I think we don't want to do this like this:

1. We'll eventually need to take action when we hit the end of store or an error, so know what gossip to ask for.  For the moment this means using the feature bit, which means we want to know this before we connect to anyone.

2. This will continue to replay messages as we get them in, which is unnecessary.